### ISSUE TITLE
simplify need to be improved

### ISSUE THREAD

**Issue by sympy-issue-migrator** [sympy/sympy#4886]
## Issue has been fixed; needs test

```
given two equation x**2+y**2=R**2 and a*x+b*y=c.
Using sage we get the result:
x=(a*c-b*sqrt((a**2+b**2)*R**2-c**2))/(a**2+b**2).  But using sympy, a very
complex result appears.

it's very luck that I write the result sympy given on the paper, and verify
that it can be transformed to the simple sage result.

So I think that the function simplify need to be improved.
```

Original issue for #4886: http://code.google.com/p/sympy/issues/detail?id=1787
Original author: https://code.google.com/u/116604158866941894488/

**Comment by divyanshu132** [comment]
@smichr I think it only needs a test or is there something which needs to be added?

**Comment by ayushbisht2001** [comment]
@asmeurer  , I would like to work on this issue. But can you please help me, where to add tests and for which function.

**Comment by 0sidharth** [comment]
This could be closed as the test case has been added in #21084
